src/object/README.md

```diff
@@ -1,261 +1,145 @@
-
-- While doubling domains, each key-range should be split into two ranges. For example, a tablet that is responsible for key range [16k-32k], after splitting, is responsible for key range [16k-24k] and its new neighbor is responsible for key range [24k-32k].
-- While doubling targets within domains, OIT will double the global key range, as shown in the <a href="#f10.10">figure</a> below. Because the key range is doubled, half of OI records map to the extended key range, and are moved to those new targets.
-
-<a id="f10.10"></a>
-
-**Object index table**
-
-![../../doc/graph/Fig_051.png](../../doc/graph/Fig_051.png "Rebalance OI table")
-
-The following <a href="#f10.11">figure</a> is an example of OIT, in this figure, OIT is a 3-way replicated KV store, keys are object IDs and values are object metadata, which include the following:
-
-- Target index for target affinity object
-- Hash stride, or hash distance between targets within the same redundancy group, see the description above of doubling domains for details
-- Metadata for dynamically striped object, which is �last extended offset� for byte array, and �split version� for KV object, more details about them are available in the section for �Dynamically Striped Object�.
-
-Although upper layers may store their metadata as object data transparently to DAOS-SR, they may also store their object metadata in the OIT. This not only takes advantage of OIT resilience but also minimizes data movement and network round trips because it can be returned on object open.
-
-<a id="f10.11"></a>
-
-**Object index table**
-
-![../../doc/graph/Fig_052.png](../../doc/graph/Fig_052.png "Object Metadata and OI table")
-
-<a id="10.4"></a>
 ## Object Schema and object class
 
-The DAOS object schema describes the definitions for object types, data protection methods, and data distribution strategies. Based on these definitions and the requirements, upper layer stacks of DAOS-SR can register a number of object classes. An object class has a unique class ID, which is a 16-bit value, and can represent a category of objects that use the same schema and schema attributes. A successfully registered object class is stored as container metadata; it is valid in the lifetime of the container. In addition, DAOS-SR also provides some pre-defined object class for the most common use.
+To avoid scaling problems and overhead common to traditional storage stack,
+DAOS objects are intentionally very simple. No default object metadata beyond
+the type and schema (object class ownership) are provided. This means that the
+system does not maintain time, size, owner, permissions and opener tracking
+attributes. All object operations are idempotent and can be processed multiple
+times with the same results. This guarantees that any operations can be repeated
+until successful or abandoned, without having to handle execute-once semantic
+and reply reconstruction.
```

Comment:
I am not sure this idempotent thing is correct anymore, especially with the new transaction model and EC.